lib/client/mfa/cli_test.go

```diff
@@ -79,25 +80,177 @@ func TestCLIPrompt(t *testing.T) {
 				},
 			},
 		}, {
-			name:         "OK webauthn or totp choose webauthn",
-			expectStdOut: "Tap any security key or enter a code from a OTP device\n",
+			name:         "OK sso",
+			expectStdOut: "", // sso stdout is handled internally in the SSO ceremony, which is mocked in this test.
+			challenge: &proto.MFAAuthenticateChallenge{
+				SSOChallenge: &proto.SSOChallenge{},
+			},
+			expectResp: &proto.MFAAuthenticateResponse{
+				Response: &proto.MFAAuthenticateResponse_SSO{
+					SSO: &proto.SSOResponse{
+						RequestId: "request-id",
+						Token:     "mfa-token",
+					},
+				},
+			},
+		}, {
+			name:         "OK prefer otp when specified",
+			expectStdOut: "Enter an OTP code from a device:\n",
+			stdin:        "123456",
+			challenge: &proto.MFAAuthenticateChallenge{
+				WebauthnChallenge: &webauthnpb.CredentialAssertion{},
+				TOTP:              &proto.TOTPChallenge{},
+				SSOChallenge:      &proto.SSOChallenge{},
+			},
+			modifyPromptConfig: func(cfg *mfa.CLIPromptConfig) {
+				cfg.PreferOTP = true
+			},
+			expectResp: &proto.MFAAuthenticateResponse{
+				Response: &proto.MFAAuthenticateResponse_TOTP{
+					TOTP: &proto.TOTPResponse{
+						Code: "123456",
+					},
+				},
+			},
+		}, {
+			name:         "OK prefer sso when specified",
+			expectStdOut: "",
+			challenge: &proto.MFAAuthenticateChallenge{
+				WebauthnChallenge: &webauthnpb.CredentialAssertion{},
+				TOTP:              &proto.TOTPChallenge{},
+				SSOChallenge:      &proto.SSOChallenge{},
+			},
+			modifyPromptConfig: func(cfg *mfa.CLIPromptConfig) {
+				cfg.PreferSSO = true
+			},
+			expectResp: &proto.MFAAuthenticateResponse{
+				Response: &proto.MFAAuthenticateResponse_SSO{
+					SSO: &proto.SSOResponse{
+						RequestId: "request-id",
+						Token:     "mfa-token",
+					},
+				},
+			},
+		}, {
+			name: "OK prefer webauthn with authenticator attachment requested",
+			expectStdOut: "" +
```

Comment:
If they chose an attachment we could skip the message, right? It shows enough intent.